src/Dialogs/Task/Manager/TaskActionsPanel.cpp

```diff
@@ -103,6 +107,23 @@ TaskActionsPanel::OnDeclareClicked()
   ExternalLogger::Declare(decl, way_points.GetHome().get());
 }
 
+inline void
+TaskActionsPanel::OnDownloadClicked()
+{
+  const ComputerSettings &settings = CommonInterface::GetComputerSettings();
+  char url[256];
+  char id[20];
+
+  strcpy(id, settings.logger.pilot_weglide_id.c_str());
+  snprintf(url, sizeof(url),"https://api.weglide.org/v1/task/declaration/%s?cup=false&tsk=true",id);
+  
+  const auto cache_path = MakeLocalPath(_T("weglide"));
+  File::Delete(LocalPath(_T("weglide/weglide_declared.tsk")));
+  Net::DownloadManager::Enqueue(url, Path(_T("weglide/weglide_declared.tsk")));
+  
+  DirtyTaskListPanel();
```

Comment:
The "Enqueue" call is asynchronous. After it finishes, the file hasn't yet been downloaded. Download might not even have been started. So why call `DirtyTaskListPanel()`?
How do you present download errors to the user? There's no error handling at all here. You don't even know if deleting the file was successful.